### ISSUE TITLE
Why my mAP is always 0?

### ISSUE THREAD

**Issue by SwimmingLiu** [ultralytics/ultralytics#1299]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Just like the below pics, AMP check is passed,but I trained it the all loss values are NAN
![image](https://user-images.githubusercontent.com/53814462/223340115-8e9035cf-b593-4336-942e-9c3154cdb96f.png)

Then I found some guys said that changing self.amp in  trainer.py can make it work

    
     self.amp = False

![image](https://user-images.githubusercontent.com/53814462/223339445-680633ce-3a06-4759-bf14-c1596341ac31.png)

After I changed this one, all loss values work correctly, but the MAP is always 0, it is so weird
![image](https://user-images.githubusercontent.com/53814462/223317651-1c12e6a2-1aeb-4093-91c9-376e1ca90881.png)

**Comment by glenn-jocher** [comment]
@SwimmingLiu it looks like your system is not compatible with AMP training since on disabling your losses return to normal.

Zero metrics are normal when starting. If you have zero mAP after 500 epochs wake me up.

**Comment by jindameias** [comment]
Maybe this is a bug, but it doesn't affect the result.

I used the cpu training shown in the top half of the figure below, which was fine.
The MAP is always 0 when I set amp=False and start training with the gpu, like the red area in the figure.
But everything is ok, the best.pt can predict accurately

![t](https://user-images.githubusercontent.com/40633372/223935636-8ece11cc-c8cc-4e51-b2c1-285b529a823f.png)

**Comment by SwimmingLiu** [comment]
@glenn-jocher  it works well on v8.0.51. GOOD JOB!!!

**Comment by zqgtba** [comment]
@SwimmingLiu @glenn-jocher Could you tell me where and how to add this key to solve the problem? Thanks a lot!

**Comment by glenn-jocher** [comment]
Certainly, @zqgtba! 

To add the `amp=True` key to the default.cfg for simpler disabling without changing the code, you can follow these steps:

1. Open the `default.cfg` file in a text editor.
2. Scroll to the `[TRAIN]` section.
3. Add a new line after the `adam` line and type `amp=True`.
4. Save the file.

By adding `amp=True` to the `[TRAIN]` section of the `default.cfg` file, AMP training will be enabled by default. 
You can then disable it by setting `amp=False` in the `train.py` file. 

I hope this helps! Let me know if you have any further questions.

**Comment by Mary14-design** [comment]
I have a slightly diffrent issue, the map values stays the same and does not change at all. I kept tuning all the hyperparameters but nothing so far even when tested on different datasets.

`  Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
        1/3      2.62G      2.404      3.162      2.083         38        640: 100%|██████████| 32/32 [00:17<00:00,  1.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 16/16 [00:11<00:00,  1.41it/s]
                   all        128        929      0.632      0.014     0.0112    0.00204

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
        2/3      2.72G      2.559       3.28      2.214         65        640: 100%|██████████| 32/32 [00:14<00:00,  2.21it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 16/16 [00:12<00:00,  1.29it/s]
                   all        128        929      0.632      0.014     0.0112    0.00204

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
        3/3       2.7G      2.541      3.333      2.374        107        640: 100%|██████████| 32/32 [00:16<00:00,  1.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 16/16 [00:12<00:00,  1.23it/s]
                   all        128        929      0.632      0.014     0.0112    0.00204`

**Comment by glenn-jocher** [comment]
Please ensure you are using the latest version of the Ultralytics package. If the issue persists, try running your training without AMP by setting `amp=False` in your configuration. If the problem continues, please provide more details about your dataset and training setup.

**Comment by Mary14-design** [comment]
I am setting the `amp=False` but still does not work and tried all the options mentioned here.

`import os
import torch
os.environ['WANDB_DISABLED'] = 'true'
from ultralytics import YOLO

if __name__ == '__main__':

    model = YOLO('yolov8s.yaml').load('yolov8s.pt')
    dataset = 'coco.yaml'
    model.train(data=dataset, epochs=150, lr0=1e-3, imgsz=480, batch=8, amp=False, device=[2]) `

`ultralytics              8.2.60`

`torch                    2.2.2+cu121
torchaudio               2.2.2+cu121
torchvision              0.17.2+cu121 `

**Comment by glenn-jocher** [comment]
@Mary14-design please ensure you are using the latest version of the Ultralytics package. If the issue persists, try running your training without AMP by setting `amp=False` in your configuration. If the problem continues, please provide more details about your dataset and training setup.

**Comment by Mary14-design** [comment]
I am setting the `amp=False` and latest version of ultralytics, tested it on coco128 and coco2017. Here is the snippet of the code 

`import os
import torch
os.environ['WANDB_DISABLED'] = 'true'
from ultralytics import YOLO

if __name__ == '__main__':

    model = YOLO('yolov8s.yaml').load('yolov8s.pt')
    #ckpt = torch.load("/workspaces/neurosoc/aihwkit/neurosoc-aihwkit/ultralytics/runs/detect/train149/weights/best.pt")
    #print("the model", model)
    dataset = 'coco128.yaml'
    model.train(data=dataset, epochs=50, lr0=1e-3, imgsz=640, batch=4, amp=False, device=[2]) `
    
    The output of the code
    
    `Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
        1/5      2.62G      2.342      3.149      2.009         38        640: 100%|██████████| 32/32 [00:09<00:00,  3.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 16/16 [00:05<00:00,  3.03it/s]
                   all        128        929    0.00463     0.0271    0.00778    0.00258

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
        2/5      2.71G      2.185      2.709      2.001         65        640: 100%|██████████| 32/32 [00:07<00:00,  4.23it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 16/16 [00:04<00:00,  3.75it/s]
                   all        128        929    0.00463     0.0271    0.00778    0.00258

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
        3/5      2.69G       2.17      2.565      1.983        107        640: 100%|██████████| 32/32 [00:07<00:00,  4.26it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 16/16 [00:04<00:00,  3.86it/s]
                   all        128        929    0.00463     0.0271    0.00778    0.00258

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
        4/5      2.72G      2.163      2.442      1.994         71        640: 100%|██████████| 32/32 [00:07<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 16/16 [00:04<00:00,  3.66it/s]
                   all        128        929    0.00463     0.0271    0.00778    0.00258

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
        5/5      2.72G      2.094      2.479      1.964         19        640: 100%|██████████| 32/32 [00:07<00:00,  4.41it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 16/16 [00:04<00:00,  3.80it/s]
                   all        128        929    0.00463     0.0271    0.00778    0.00258`

**Comment by glenn-jocher** [comment]
It seems like your mAP values are not changing. Please ensure your dataset annotations are correct and that your model is learning effectively. If the issue persists, try using a different dataset or adjusting your hyperparameters further.
